frontend/apps/design-system-storybook/.storybook/main.ts

```diff
@@ -0,0 +1,73 @@
+import { join, dirname, resolve } from "node:path";
+import {StorybookConfig} from "@storybook/react-webpack5";
+
+/**
+ * This function is used to resolve the absolute path of a package.
+ * It is needed in projects that use Yarn PnP or are set up within a monorepo.
+ */
+function getAbsolutePath(value) {
+  return dirname(require.resolve(join(value, "package.json")));
+}
+
+const config: StorybookConfig = {
+  stories: [
+    "../../../packages/dsco/src/**/*.story.@(ts|tsx)"
+  ],
+  addons: [
+    getAbsolutePath("@storybook/addon-webpack5-compiler-swc"),
```

Comment:
So we retain a webpack dependency for storybook? How do we feel about that?

Reply:
Let's chat about this at standup, the initial thinking is that we still need to compile SASS and we can do it in a number of ways, including using the more modern [vite](https://storybook.js.org/docs/get-started/frameworks/react-vite) framework or perhaps the [nextjs](https://storybook.js.org/docs/get-started/frameworks/nextjs) framework. 

Vite would get rid of the dependency on webpack and we'd have to do sass compilation via vite instead. Next.js actually uses webpack under the hood so we'd still need to configure it to compile sass for us. 

The webpack config is pretty lightweight though, it's the styling and `webpackFinal` parts below to allow sass compilation and provide pathing. However we can swap to another framework if that works better for us. In giving consideration to the lightweight webpack config and our continued usage of webpack in Next.js and dashboard I figured using the webpack framework made the most sense. That being said, we can swap to a different framework pretty easily in the future (compared to our current storybook) by swapping out the `framework` line in this file and reconfiguring the configs below.

Reply:
I'm fine with leaving Webpack here (seems like it'd be unlikely that our config would get particularly complicated for Storybook), but definitely interested to learn more about other options for more complicated applications (I've definitely experienced the complexity of our `apps` Webpack config!)